### PR TITLE
Let iOS have a minimum scroll movement threshold to break before motion starts

### LINKED ISSUE #12181: After drag and stop, finger lift can still cause a small position shift
Seems to happen on Android and iOS.

After a drag and a complete stop, the finger lifting motion can still cause a small movement which happens more or less with the same frequency on native Android but seems mostly suppressed on native iOS. 

Match on iOS with a filter after stopped scrolls.

### PR DESCRIPTION
**Pull request by xster**
Fixes #12181

Mimics native behaviour. Android unaffected.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scroll_activity.dart [hunk 1]
```diff
@@ -227,12 +227,18 @@ class ScrollDragController implements Drag {
     @required DragStartDetails details,
     this.onDragCanceled,
     this.carriedVelocity,
+    this.motionStartDistanceThreshold
   }) : assert(delegate != null),
        assert(details != null),
+       assert(
+         motionStartDistanceThreshold == null || motionStartDistanceThreshold > 0.0,
+         'Must be a positive number or null'
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: This should say _what_ must be positive or null. The error message will have no context.

File: packages/flutter/lib/src/widgets/scroll_activity.dart [hunk 2]
```diff
@@ -227,12 +227,18 @@ class ScrollDragController implements Drag {
     @required DragStartDetails details,
     this.onDragCanceled,
     this.carriedVelocity,
+    this.motionStartDistanceThreshold
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: nit: trailing comma

File: packages/flutter/lib/src/widgets/scroll_activity.dart [hunk 3]
```diff
@@ -265,22 +283,66 @@ class ScrollDragController implements Drag {
...
+    }
+  }
+
+  /// If a motion start threshold exists, determine whether the threshold is
+  /// reached to start applying position offset.
+  ///
+  /// Returns false either way if there's no offset.
+  bool _breakMotionStartThreshold(double offset, Duration timestamp) {
+    if (timestamp == null) {
+      // If we can't track time, we can't apply thresholds. Should never happen.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_activity.dart]: If it should never happen, assert instead?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/widgets/scroll_activity.dart (hunk 1 above)
Comment (xster) [reply packages/flutter/lib/src/widgets/scroll_activity.dart]: Seems like asserts with optional messages actually shows both the assert test code and the message.

Comment (Hixie) [reply packages/flutter/lib/src/widgets/scroll_activity.dart]: Yes, but experience shows our users don't understand the assertion test code. They only read the text (if we're lucky). (The framework actually flips the message around so the message comes first and the assert code is second, now.)

Comment (xster) [reply packages/flutter/lib/src/widgets/scroll_activity.dart]: Done

File: packages/flutter/lib/src/widgets/scroll_activity.dart (hunk 2 above)
Comment (xster) [reply packages/flutter/lib/src/widgets/scroll_activity.dart]: Done

File: packages/flutter/lib/src/widgets/scroll_activity.dart (hunk 3 above)
Comment (xster) [reply packages/flutter/lib/src/widgets/scroll_activity.dart]: Clarified comment. Can be null with accessibility gestures

**Pull request merged**